### PR TITLE
Improve Library Display performance and Refactor grid viewmodels

### PR DESCRIPTION
**Pull request by Mbucari**
# [Improve Library load and refresh performance](https://github.com/rmcrackan/Libation/commit/e1cd8b8f940ebec33c6c6a25b68a9a69cf5473e1)

- Writing library scans to zip file is now async
- Fixed the n^3 grid refresh problem
- Fixed another grid refresh bug on Avalonia
- Force garbage collection after a library scan. GC was blowing up my rap, each scan adding ~300MB. IDK why it wasn't collection on its own. Wait, maybe because I now have 64GB or ram and there's absolutely no reason to GC when I'm nowhere near capacity. lol

# [WinForms and Avalonia now share all GridEntry view models](https://github.com/rmcrackan/Libation/commit/fb9d0625459fc8c03fce017188086269af1d4d3a)

This is the big one. I successfully merged WinForms and Avalonia GridEntery view models and moved them into LibationUiBase. This removed over 600 lines of code.

I'd been working on this one for a while, and fixing #525 really threw a wrench into this.
This will make maintenance a LOT easier.

### CODE REVIEW

**Comment by rmcrackan** [comment]
> GridEntery ... into LibationUiBase. This removed over 600 lines of code. ... This will make maintenance a LOT easier.

Thank you!!! This is exactly how I hoped LibationUiBase would eventually get used.

File: Source/ApplicationServices/LibraryCommands.cs [hunk 1]
```diff
@@ -166,14 +167,15 @@ public static async Task<(int totalCount, int newCount)> ImportAccountAsync(Func
...
+                GC.Collect(GC.MaxGeneration, GCCollectionMode.Aggressive, true, true);
             }
         }
 
         private static async Task<List<ImportItem>> scanAccountsAsync(Func<Account, Task<ApiExtended>> apiExtendedfunc, Account[] accounts, LibraryOptions libraryOptions)
         {
             var tasks = new List<Task<List<ImportItem>>>();
 
-            using LogArchiver archiver
+            await using LogArchiver archiver
```
Comment (rmcrackan) [inline Source/ApplicationServices/LibraryCommands.cs:178]: I have never seen not contemplated `await using`; that's wild. What's being awaited? Isn't this just a ctor?

File: Source/FileManager/LogArchiver.cs [hunk 2]
```diff
@@ -6,10 +6,11 @@
 using System.IO.Compression;
 using System.Linq;
 using System.Text;
+using System.Threading.Tasks;
 
 namespace FileManager
 {
-	public class LogArchiver : IDisposable
+	public sealed class LogArchiver : IAsyncDisposable
```
Comment (rmcrackan) [inline Source/FileManager/LogArchiver.cs:13]: Ah, I've never seen IAsyncDisposable. Answers my previous question.

File: Source/ApplicationServices/LibraryCommands.cs (hunk 1 above)
Comment (Mbucari) [reply Source/ApplicationServices/LibraryCommands.cs:178]: It's actually a waiting [DisposeAsync](https://learn.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-disposeasync).

ZipArchive writes all changes to disk when it's disposed, so it's good to make it async.

**Comment by Mbucari** [comment]
> > GridEntery ... into LibationUiBase. This removed over 600 lines of code. ... This will make maintenance a LOT easier.
> 
> Thank you!!! This is exactly how I hoped LibationUiBase would eventually get used.

You're very welcome. Definitely took quite a bit of work, and there is a tiny bit of redundancy when it comes to invoking PropertyChanged so that all bases are covered for both avalonia and wind forms, but on the whole I think it worked out rather nicely.

**Pull request merged**